### PR TITLE
adding / updating linux / android rules

### PR DESCRIPTION
**Pull request by mike-hunhoff**

### CODE REVIEW

File: host-interaction/process/create/create-process-on-linux.yml
Comment (williballenthin) [inline host-interaction/process/create/create-process-on-linux.yml:25]: which syscall is this?

File: nursery/execute-syscall.yml
Comment (williballenthin) [inline nursery/execute-syscall.yml:17]: can you add a comment for this? what function/runtime?

File: nursery/truncate-file-on-linux.yml
Comment (williballenthin) [inline nursery/truncate-file-on-linux.yml:10]: os filter maybe?

**Review by williballenthin — APPROVED** [review]
few small things, mainly descriptions, then ready to go!

File: host-interaction/file-system/write/write-file-on-linux.yml
Comment (mr-tz) [inline host-interaction/file-system/write/write-file-on-linux.yml:32]: also vfprintf?

File: host-interaction/file-system/write/write-file-on-linux.yml
Comment (mr-tz) [inline host-interaction/file-system/write/write-file-on-linux.yml:31]: also vdprintf?

File: nursery/modify-api-blacklist-or-denylist-via-jni-on-android.yml
Comment (mr-tz) [inline nursery/modify-api-blacklist-or-denylist-via-jni-on-android.yml:17]: is this string required/useful here?

**Review by mr-tz — COMMENTED** [review]
nice work, a few items to discuss inline

File: nursery/modify-api-blacklist-or-denylist-via-jni-on-android.yml
Comment (mike-hunhoff) [reply nursery/modify-api-blacklist-or-denylist-via-jni-on-android.yml:17]: It's used as part of the JNI invocation - I added it to knock down the FP potential but it'd probaly be fine w/out. Thoughts?

File: host-interaction/process/create/create-process-on-linux.yml
Comment (mike-hunhoff) [reply host-interaction/process/create/create-process-on-linux.yml:25]: Good point 😄 (`execve`) - updated

File: host-interaction/file-system/write/write-file-on-linux.yml
Comment (mike-hunhoff) [reply host-interaction/file-system/write/write-file-on-linux.yml:31]: added

File: host-interaction/file-system/write/write-file-on-linux.yml
Comment (mike-hunhoff) [reply host-interaction/file-system/write/write-file-on-linux.yml:32]: added

File: nursery/execute-syscall.yml
Comment (mike-hunhoff) [reply nursery/execute-syscall.yml:17]: It's https://man7.org/linux/man-pages/man2/syscall.2.html. I'm not sure what you're asking here. I can add `os` (`linux`, `android`)?

File: nursery/truncate-file-on-linux.yml
Comment (mike-hunhoff) [reply nursery/truncate-file-on-linux.yml:10]: added

**Comment by mike-hunhoff** [comment]
@williballenthin @mr-tz we're failing lints because `aarch64` isn't introduced until https://github.com/mandiant/capa/tree/feat/1755. That means we can't merge this until the referenced PR is merged?

File: nursery/execute-syscall.yml
Comment (williballenthin) [reply nursery/execute-syscall.yml:17]: sorry I wasn't more clear. 

Yeah I wasn't sure if it was some GCC built in, or a linux thing, or something from another platform, etc. So, I think a comment or description would help. Also constraining the OS makes sense too, since it's Linux specific.

**Comment by williballenthin** [comment]
hm let's pull the aarch64 and android features into master now so that we can support these rules. I'm afraid of too many things getting blocked behind #1755

File: nursery/modify-api-blacklist-or-denylist-via-jni-on-android.yml
Comment (mr-tz) [reply nursery/modify-api-blacklist-or-denylist-via-jni-on-android.yml:17]: let's leave it and maybe add a comment

**Review by mr-tz — APPROVED** [review]

**Comment by mr-tz** [comment]
It may just be this commit: https://github.com/mandiant/capa/commit/e1186c5b4b30ea477e51bef144413f90b7f72798 or we can take all the changes to `capa/features/common.py`.

**Pull request merged**
